### PR TITLE
Preliminary code structure docs

### PR DESCRIPTION
**Pull request by davidjoy**
"What are all these directories?"

"Read this!"

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -50,6 +50,39 @@ Assuming you have a fairly standard devstack setup, if you click the "Upgrade to
...
+
+The source for this project is organized into nested submodules according to the ADR `Feature-based Application Organization <https://github.com/edx/frontend-cookiecutter-application/blob/master/docs/decisions/0002-feature-based-application-organization.rst>`_.
+
+Breakdown of the ``src`` directory:
+
+**assets**
+  Image assets used by the top-level code.
+
+**common**
+  Boilerplate code that is common to many of our frontend applications.  Currently copied from place to place, it is intended to live in `edx/frontend-common <https://github.com/edx/frontend-common>`_.
```
Comment (albemarle) [inline README.rst]: Maybe "to eventually live".

File: README.rst [hunk 2]
```diff
@@ -50,6 +50,39 @@ Assuming you have a fairly standard devstack setup, if you click the "Upgrade to
...
+  Boilerplate code that is common to many of our frontend applications.  Currently copied from place to place, it is intended to eventually live in `edx/frontend-common <https://github.com/edx/frontend-common>`_.
+
+**components**
+  Top-level App.jsx component, which is 95% shared across frontends and will eventually get similar treatment to the ``common`` directory.
+
+**data**
+  Top-level redux/redux-saga reducers and sagas.
+
+**feedback**
+  A reusable component which displays user feedback messages as alerts at the top of the page.  While it is currently only in use by this application, it's intended to be generic and shared across applications, so should remain free of payment-specific code.  It will eventually live in either `edx/paragon <https://github.com/edx/paragon>`_ or its own repo.
```
Comment (albemarle) [inline README.rst]: As a note, I haven't heard these called feedback messages anywhere else.  Terms I've heard are server messages, user notifications, or alerts.  Paragon calls them StatusAlert, but that's more what the UI element is called.

File: README.rst [hunk 3]
```diff
@@ -50,6 +50,39 @@ Assuming you have a fairly standard devstack setup, if you click the "Upgrade to
...
+
+**i18n**
+  The language configuration for the app.
+
+**payment**
+  The guts of this app.  This includes all payment forms, payment methods, order details, data models, and associated API calls.
+
+  Please see src/payment/README.rst for more detail.
```
Comment (albemarle) [inline README.rst]: Could make this a link if you're feeling super nice.  :)

File: README.rst (hunk 2 above)
Comment (davidjoy) [reply README.rst]: Yeah, I'm not crazy about the name either.  May fix if it ever makes its way out of this repo. :)

File: README.rst (hunk 3 above)
Comment (davidjoy) [reply README.rst]: There's nowhere to link it to yet since this is in a branch... ;)

Comment (albemarle) [reply README.rst]: Pfft.  Live life boldly!

**Pull request merged**